### DISCUSSION TITLE
GFW of China or Iran? Need 3rd-party server or not?

### DISCUSSION THREAD

**Discussion by UjuiUjuMandan** [GFW-knocker/gfw_resist_tls_proxy#59] [Q&A]
Questions.

1. Does this tool work in China or not?
2. Does it need a 3rd-party to relay traffic, or it just work between website and client?

**Comment by GFW-knocker** [comment]
1.  from its behavior we guess that GFW is the same in both country.
2. yes you need VPS+websocket config behind Cloudflare IP
3. in future , we may find a way to apply this tools directly on some popular site.

**Comment by free-the-internet** [comment]
> Questions.
> 
> 1. Does this tool work in China or not?
> 2. Does it need a 3rd-party to relay traffic, or it just work between website and client?

I think in China Cloudflare has not any problem and you can use your own server's SNI.
Only if you are using workers method, and GFW drops Cloudflare worker domain, you need this tool.
Finally, using this tool, maybe you can also open the websites and services that only their domain is blacklisted (and not their IP)
